### PR TITLE
[MRG+1] Set hexbin default linecolor to 'face'

### LINKED ISSUE #7185: Hexbin default edgecolors kwarg is misnamed
Hexbin has a default 'edgecolors' kwarg value of 'none', which it internally translates to 'face'.  This is highly confusing and inconsistent, and it blocks the user from specifying 'none'.  Granted, the 'none' value may be undesirable because of patch boundary artifacts, but it should not be interpreted as a completely different value.
The 'face' value is used to prevent artifacts by adding boundary lines.  hexbin sets the default 'linewidths' to 1, which is probably much larger than it needs to be for this purpose, resulting in prominent artifacts when the number of hexagons is large.

### PR DESCRIPTION
**Pull request by dstansby**
Fixes #7185

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -4163,10 +4163,12 @@ def hexbin(self, x, y, C=None, gridsize=100, bins=None,
         linewidths : scalar, optional, default is *None*
             If *None*, defaults to 1.0.
 
-        edgecolors : {'none'} or mpl color, optional, default is 'none'
-            If 'none', draws the edges in the same color as the fill color.
-            This is the default, as it avoids unsightly unpainted pixels
-            between the hexagons.
+        edgecolors : {'face', 'none', *None*} or mpl color, optional, default is 'face'
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: This line is too long for pep8. It should be 79 char or less.

Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: This is an acceptable place to use `\` to break the doc line.

File: doc/users/dflt_style_changes.rst [hunk 2]
```diff
@@ -560,6 +560,13 @@ or by setting::
 
 in your :file:`matplotlibrc` file.
 
+``hexbin``
+----------
+
+The default value of the ``linecolor`` kwarg for `~matplotlib.Axes.hexbin` has
+changed from ``'none'`` to ``'face'``. If 'none' is now supplied, no line edges
+are drawn around the hexagons.
```
Comment (tacaswell) [inline doc/users/dflt_style_changes.rst]: This should note that the default _behavior_ has not changed (just the meaning of `'none'`).

**Comment by efiring** [comment]
On 2016/11/23 10:34 AM, Thomas A Caswell wrote:
> *@tacaswell* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In lib/matplotlib/axes/_axes.py
> <https://github.com/matplotlib/matplotlib/pull/7500>:
>
>> @@ -4163,10 +4163,12 @@ def hexbin(self, x, y, C=None, gridsize=100, bins=None,
>          linewidths : scalar, optional, default is *None*
>              If *None*, defaults to 1.0.
>
> -        edgecolors : {'none'} or mpl color, optional, default is 'none'
> -            If 'none', draws the edges in the same color as the fill color.
> -            This is the default, as it avoids unsightly unpainted pixels
> -            between the hexagons.
> +        edgecolors : {'face', 'none', *None*} or mpl color, optional, default is 'face'
>
> This is an acceptable place to use |\| to break the doc line.

Or just leave out the "optional"?  After all, specifying the default 
only makes sense if it is optional.
Alternatively, a reasonable convention would be to say that if the 
default is not specified on an optional parameter, the default is the
first entry in the list of possibilities.

**Comment by QuLogic** [comment]
> the default is the first entry in the list of possibilities.

That's how Numpydoc already defines arguments that take a restricted set of inputs.

File: lib/matplotlib/axes/_axes.py (hunk 1 above)
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: something like
edgecolors : {"face", "none", None, color}, optional, default: "face" would also have been acceptable.

**Comment by NelleV** [comment]
Thanks for the patch @dstansby !

**Pull request merged**